src/api/v3/pages.py

```diff
@@ -36,10 +36,20 @@ def transform_page(page_translation):
 def pages(request, region_slug, language_code):
     region = Region.get_current_region(request)
     result = []
-    for page in region.pages.all():
+    for page in Page.objects.filter(region=region, parent=None):  # get main level
         page_translation = page.get_public_translation(language_code)
         if page_translation:
             result.append(transform_page(page_translation))
+            result = get_children(region, page, language_code, result)
     return JsonResponse(
         result, safe=False
     )  # Turn off Safe-Mode to allow serializing arrays
+
+
+def get_children(region, parent, language_code, result):
+    for page in Page.objects.filter(region=region, parent=parent):
```

Comment:
```suggestion
    for page in parent.children.filter(archived=False):
```
- `children` is the reverse related name of `parent` and thus contains all direct children
- filter for region is not required because the children always have the same region as the parent
- exclude archived children